modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -106,5 +118,92 @@ object ConstructionService {
       } else {
         parseUnsignedTransaction(hex)
       }
+
+    def combineTransaction(hex: Hex, signatures: NonEmptyList[RosettaSignature]): EitherT[F, Throwable, Either[ConstructionError, Hex]] =
+      KryoSerializer[F]
+        .deserialize[Transaction](hex.toBytes)
+        .traverse { transaction =>
+          convertSignature(signatures).map { test2 =>
+            test2.left
+              .map(err => Left(err): Either[ConstructionError, Hex])
+              .map { signatureProofs =>
+                val serializedTransaction: Either[ConstructionError, Hex] = KryoSerializer[F]
+                  .serialize(
+                    Signed[Transaction](
+                      transaction,
+                      NonEmptySet(signatureProofs.head, SortedSet(signatureProofs.tail: _*))(
+                        Order.fromOrdering(SignatureProof.OrderingInstance)
+                      )
+                    )
+                  )
+                  .left
+                  .map(_ => MalformedTransaction)
+                  .map(bytes => Hex.fromBytes(bytes))
+
+                serializedTransaction
+              }
+              .merge
+          }
+        }
+        .map { test =>
+          test.left.map(_ => Left(MalformedTransaction): Either[ConstructionError, Hex]).merge
+        }
+        .attemptT
+
+    def convertSignature(
+      signature: NonEmptyList[RosettaSignature]
+    ) = {
```

Comment:
use plural name as it takes a list of signatures, not a single signature

Reply:
Changed to `signatures`.